js/Common.js

```diff
@@ -395,6 +395,35 @@ if (!CRM.vars) CRM.vars = {};
     return ret + _.escape(row.text) + (description ? '<div class="crm-select2-row-description"><p>' + _.escape(description) + '</p></div>' : '');
   }
 
+  /**
+   * Helper to generate an icon with alt text.
+   *
+   * See also smarty `{icon}` and CRM_Core_Page::crmIcon() functions
+   *
+   * @param string icon
+   *   The Font Awesome icon class to use.
+   * @param string text
+   *   Alt text to display.
+   * @param mixed condition
+   *   This will only display if this is truthy.
+   *
+   * @return string
+   *   The formatted icon markup.
+   */
+  CRM.utils.formatConditionalIcon = function (icon, text, condition) {
+    if (!condition) {
+      return '';
+    }
```

Comment:
Taking a leaf from #17307 how about making the condition optional?

```suggestion
  CRM.utils.formatIcon = function (icon, text, condition) {
    if (typeof condition !== 'undefined' && !condition) {
      return '';
    }
```

Reply:
Great suggestion--I just added a commit that does this and changes the name where it's used